modules/busco/main.nf

```diff
@@ -0,0 +1,83 @@
+process BUSCO {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::busco=5.3.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/busco:5.3.2--pyhdfd78af_0':
+        'quay.io/biocontainers/busco:5.3.2--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(fasta, stageAs: 'tmp_input/*')  // Required:    meta map, and fasta sequence files
+    each lineage                                          // Required:    lineage to check against
+    path busco_lineages_path                              // Recommended: path to busco lineages - downloads if not set
+    path config_file                                      // Optional:    busco configuration file
+
+    output:
+    tuple val(meta), path("*-busco.batch_summary.txt"), emit: batch_summary
+    tuple val(meta), path("short_summary.*.txt")      , emit: short_summaries_txt, optional: true
+    tuple val(meta), path("short_summary.*.json")     , emit: short_summaries_json, optional: true
+    tuple val(meta), path("*-busco")                  , emit: busco_dir
+    path "versions.yml"                               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}-${lineage}"
+    def busco_config = config_file ? "--config $config_file" : ''
+    def busco_lineage_dir = busco_lineages_path ? "--offline --download_path ${busco_lineages_path}" : ''
+    """
+    # Nextflow changes the container --entrypoint to /bin/bash (container default entrypoint: /usr/local/env-execute)
+    # Check for container variable initialisation script and source it.
+    if [ -f "/usr/local/env-activate.sh" ]; then
+        set +u  # Otherwise, errors out because of various unbound variables
+        . "/usr/local/env-activate.sh"
+        set -u
+    fi
+
+    # If the augustus config directory is not writable, then copy to writeable area
+    if [ ! -w "\${AUGUSTUS_CONFIG_PATH}" ]; then
+        # Create writable tmp directory for augustus
+        AUG_CONF_DIR=\$( mktemp -d -p \$PWD )
+        cp -r \$AUGUSTUS_CONFIG_PATH/* \$AUG_CONF_DIR
+        export AUGUSTUS_CONFIG_PATH=\$AUG_CONF_DIR
+        echo "New AUGUSTUS_CONFIG_PATH=\${AUGUSTUS_CONFIG_PATH}"
+    fi
+
+    # Ensure the input is uncompressed
+    INPUT_SEQS=input_seqs
+    mkdir "\$INPUT_SEQS"
+    cd "\$INPUT_SEQS"
+    for FASTA in ../tmp_input/*; do
+        if [ "\${FASTA##*.}" == 'gz' ]; then
+            gzip -cdf "\$FASTA" > \$( basename "\$FASTA" .gz )
+        else
+            ln -s "\$FASTA" .
+        fi
+    done
+    cd ..
+
+    busco \\
+        --cpu $task.cpus \\
+        --in "\$INPUT_SEQS" \\
+        --out ${prefix}-busco \\
+        --lineage_dataset $lineage \\
```

Comment:
That's a good point. I'll push some changes though so we can run both in the same command instead.

Reply:
Grr. Stuff blowing up maybe because of space issues running the test, and also linting stuff. 
Will fix after lunch.

I've chosen to use a keyword 'auto' to trigger using `--auto-lineage` so the output is named using `auto` e.g. 'test-auto-busco`. It means specific lineages and 'auto' can be run in the same workflow. Added it to a test too.